### PR TITLE
Avoid indefinite growth of the db

### LINKED ISSUE #161: Database should not grow too big
The database is currently too big and as a result, the registration round might not be possible to be executed after startup.

We could use IPFS to store the markets to avoid this issue.

### PR DESCRIPTION
**Pull request by Adamantios**
Avoids indefinite db growth by storing the bets in the agent's local storage as a JSON file.

Fixes #161. However, it does not use SQLite as suggested in the issue and simply stores the JSON file as is.

### CODE REVIEW

File: packages/valory/skills/decision_maker_abci/behaviours/sampling.py [hunk 1]
```diff
@@ -44,22 +40,21 @@ class SamplingBehaviour(DecisionMakerBaseBehaviour):
     @property
     def available_bets(self) -> Iterator[Bet]:
         """Get an iterator of the unprocessed bets."""
-        bets = self.synchronized_data.bets
 
         # Note: the openingTimestamp is misleading as it is the closing timestamp of the bet
         if self.params.using_kelly:
             # get only bets that close in the next 48 hours
```
Comment (jmoreira-valory) [inline packages/valory/skills/decision_maker_abci/behaviours/sampling.py:46]: comment is misleading

**Review by jmoreira-valory — APPROVED** [review]

File: packages/valory/skills/market_manager_abci/rounds.py [hunk 2]
```diff
@@ -61,18 +59,13 @@ def _get_deserialized(self, key: str) -> DeserializedCollection:
...
-        """Get the most voted bets."""
-        serialized_bets = str(self.db.get("bets", ""))
-
-        if serialized_bets == "":
-            return []
-
-        return json.loads(serialized_bets, cls=BetsDecoder)
+    def bets_hash(self) -> str:
+        """Get the most voted bets' hash."""
+        return str(self.db.get_strict("bets"))
```
Comment (DavidMinarsch) [inline packages/valory/skills/market_manager_abci/rounds.py:64]: should be
```suggestion
        return str(self.db.get_strict("bets_hash"))
```
no?

File: packages/valory/skills/market_manager_abci/behaviours.py [hunk 3]
```diff
@@ -39,15 +45,54 @@
...
+    def store_bets(self) -> None:
+        """Store the bets to the agent's data dir as JSON."""
+        serialized = serialize_bets(self.bets)
+        if serialized is None:
+            return
+
+        with open(self.bets_filepath, WRITE_MODE) as bets_file:
+            bets_file.write(serialized)
+
+    def read_bets(self) -> None:
```
Comment (DavidMinarsch) [inline packages/valory/skills/market_manager_abci/behaviours.py:71]: How are you handling errors here?

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

File: packages/valory/skills/decision_maker_abci/behaviours/sampling.py (hunk 1 above)
Comment (Adamantios) [reply packages/valory/skills/decision_maker_abci/behaviours/sampling.py:46]: Please propose an alternative.

File: packages/valory/skills/market_manager_abci/behaviours.py (hunk 3 above)
Comment (Adamantios) [reply packages/valory/skills/market_manager_abci/behaviours.py:82]: Added error handling in e3ffdc533881069422e42b280ca1217ca60589a2.

**Review by 0xArdi — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**